### ISSUE TITLE
[7.0+] Avoid assertInstanceOf() and similar for explicitly cases

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#94]
For PHP 7.0, you can typehint function returns with the returned classes. Not make senses you create an `static::assertInstanceOf(Return::class, $object->getReturnClass())` when this method explicitly returns a `Return` instance, because PHP will check it first then phpunit.

```php
class Return {
    // Note that it returns "Return" on typehint.
    public static function getReturnClass(): Return {
        return new static;
    }
}

class ReturnTest {
    public function testGetReturnClass() {
        // Warning: tested method explicitly returns a Return class.
        static::assertInstanceOf(Return::class, Return::getReturnClass());
    }
}
```

It should not consider any annotation, only for the PHP 7.0+ typehint.

**This should be valid for this cases too (and negatives):**

```php
static::assertEmpty(typehintVoid());
static::assertNull(typehintVoid());
static::assertInternalType('string', typehintString());
static::assertInternalType('int', typehintInt());
static::assertInternalType('float', typehintFloat());
static::assertInternalType('bool', typehintBool());
static::assertInternalType('array', typehintArray());
```

For what I remember, is that.

**Comment by Fleshgrinder** [comment]
When implementing this be sure to take Nullables (`?Return`) into account.

**Comment by kalessil** [comment]
Picking up for 2.3.9.

Note for myself: consider only implicitly defined return types (assume PhpDoc lies)

**Comment by rentalhost** [comment]
There some way to we get the type easily on your package? The `getTypes()` don't help sometimes. It will be useful to resolve this issue.

**Comment by kalessil** [comment]
There is no easy way - PS trust PhpDoc, so leave this case for me.

**Comment by kalessil** [comment]
Finally, I'm on this lovely ticket =)

**Comment by kalessil** [comment]
Implemented
